### ISSUE TITLE
segmentation fault when running eco tracker

### ISSUE THREAD

**Issue by zanazakaryaie** [rockkingjy/OpenTracker#24]
Hello

Thanks for your great library.

I've just compiled and run both KCF and DSST tracker successfully. I expected DSST to be faster than KCF as it uses MOSSE tracker but results show that KCF is faster!

Anyway

I tried to run ECO but it gives me segmentation fault (core dumped):

```
1 205 151 17 50 
../sequences/Crossing/img/0001.jpg
max_score_threshhold: 0.100000
0:0.970590; 1:0.980296; 2:0.990099; 3:1.000000; 4:1.010000; 5:1.020100; 6:1.030301; 
Segmentation fault (core dumped)
```

**Comment by rockkingjy** [comment]
have you make install it?

**Comment by zanazakaryaie** [comment]
Yes. In terminal I wrote:`
sudo make install`

**Comment by rockkingjy** [comment]
in runecotracker.cc, change     parameters.useCnFeature = false; and make, how about it?

**Comment by zanazakaryaie** [comment]
I guessed that might be the reason. Changed it to false but the same error reproduced :(

**Comment by rockkingjy** [comment]
your system is? and makefile all set the same as in readme?

**Comment by zanazakaryaie** [comment]
OS is Ubuntu 16.04 with OpenCV 3.4.1 on my Laptop
I didn't change the makefile

**Comment by rockkingjy** [comment]
in eco/makefile, set: USE_SIMD=0, and make in the folder eco/ make install and try, how about it?

**Comment by zanazakaryaie** [comment]
I did it at now it gives me this error after make:

```
feature_extractor.o: In function `eco::FeatureExtractor::get_hog_features_simd(std::vector<cv::Mat, std::allocator<cv::Mat> >)':
/home/zana/OpenTracker/eco/feature_extractor.cc:235: undefined reference to `gradMag(float*, float*, float*, int, int, int, bool)'
/home/zana/OpenTracker/eco/feature_extractor.cc:278: undefined reference to `fhog(float*, float*, float*, int, int, int, int, int, float)'
collect2: error: ld returned 1 exit status
make: *** [runecotracker.bin] Error 1
```

**Comment by rockkingjy** [comment]
I don't understand your meaning, with USE_SIMD=0, you get this error?

**Comment by zanazakaryaie** [comment]
Yes. I changed USE_SIMD=1 to USE_SIMD=0 in makefile. Then I make but it didn't compiled because of that error I mentioned. If you look at the error, it says that get_hog_features_simd is the source of error.

**Comment by rockkingjy** [comment]
have you do `cd eco/` and then make in eco/, not in the OpenTracker/ root folder as the readme said? And the USE_SIMD=0 should be changed in OpenTracker/eco/makefile not OpenTracker/makefile.

**Comment by zanazakaryaie** [comment]
Yes I did all these stuffs in OpenTracker/eco/

**Comment by rockkingjy** [comment]
it should not be like that, USE_SIMD=0 means it will not compile and recognize `eco::FeatureExtractor::get_hog_features_simd`, you could check it in the code.

**Comment by rockkingjy** [comment]
anyway, try `make clean` and make again.

**Comment by zanazakaryaie** [comment]
Got it
Thank you
